packages/ec-linear/src/mutator/with_one_over_length.rs

```diff
@@ -10,20 +10,27 @@ use crate::genome::Linear;
 
 pub struct WithOneOverLength;
 
+#[derive(Debug, thiserror::Error, Diagnostic)]
+#[error("Failed to convert genome size {0} to an f32 value to be used as the mutation rate")]
+#[diagnostic(help = "Make sure the genome size is representable by an f32 when using this mutator")]
+pub struct GenomeSizeConversionError(usize);
+
 impl<T> Mutator<Vec<T>> for WithOneOverLength
 where
     T: Not<Output = T>,
 {
-    fn mutate(&self, genome: Vec<T>, rng: &mut ThreadRng) -> Result<Vec<T>> {
-        let genome_length = genome.len().to_f32().with_context(|| {
-            format!(
-                "The genome length {} couldn't be converted to an f32 value",
-                genome.len()
-            )
-        })?;
+    type Error = GenomeSizeConversionError;
+
+    fn mutate(&self, genome: Vec<T>, rng: &mut ThreadRng) -> Result<Vec<T>, Self::Error> {
+        let genome_length = genome
+            .len()
+            .to_f32()
+            .ok_or(GenomeSizeConversionError(genome.len()))?;
         let mutation_rate = 1.0 / genome_length;
         let mutator = WithRate::new(mutation_rate);
-        mutator.mutate(genome, rng)
+        mutator
+            .mutate(genome, rng)
+            .map_err(|_: Infallible| unreachable!())
```

Comment:
This seems rather subtle to me, and perhaps deserving of a comment? I know I had to stop and think about it for a second.